browser/android/preferences/brave_pref_service_bridge.cc

```diff
@@ -0,0 +1,48 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/build/android/jni_headers/BravePrefServiceBridge_jni.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/profiles/profile_manager.h"
+
+using base::android::JavaParamRef;
+using brave_shields::ControlType;
+
+namespace {
+
+Profile* GetOriginalProfile() {
```

Comment:
@SergeyZhukovsky where is the code you already wrote to handle this stuff?

Reply:
We set it via preferences in Java currently and we checked the preferences in OnBeforeURLRequest in chrome_network_delegate.cc, but that code is gone right now, @samartnik is moving it to another place.